File: 02Algorithm_Study/03_week_homeWork/Path.py
def solution(m, n, puddles):
    myMap = [[0 for i in range(m + 1)] for j in range(n + 1)]
    visit = myMap[:]
    visit[1][1] = 1

    for mul in puddles:
        myMap[mul[1]][mul[0]] = 1

    for x in range(1, n + 1):
        for y in range(1, m + 1):
            if(x !=1 or y != 1):
                if(myMap[x][y] == 1):
                    visit[x][y] = 0
                else:
                    visit[x][y] = (visit[x-1][y] + visit[x][y-1])%1000000007
    
    answer = visit[n][m]
    return answer


print(solution(4,3,[[2, 2]]))

File: 02Algorithm_Study/03_week_homeWork/Path.cpp
#include <string>
#include <iostream>
#include <vector>
int myMap[110][110];
int visit[110][110];
using namespace std;

int solution(int m, int n, vector<vector<int>> puddles) {
    int answer = 0;
    visit[1][1] = 1;
    for(int i = 0; i <puddles.size(); i++){
        myMap[puddles[i][1]][puddles[i][0]] = 1;
    }

    for(int i = 1; i <=n; i++){
        for(int j = 1; j <=m; j++){
            if(i !=1 || j !=1){
                if(myMap[i][j] == 1){
                    visit[i][j] = 0;
                }
                else{
                    visit [i][j] = (visit[i-1][j] + visit[i][j-1])%1000000007;
                }
            }
        }
    }
    answer = visit[n][m];
    return answer;
}

int main(){
    cout << solution(4,3,{{2,2}});
}